compose.yaml

```
@@ -0,0 +1,85 @@
+# Comments are provided throughout this file to help you get started.
+# If you need more help, visit the Docker Compose reference guide at
+# https://docs.docker.com/go/compose-spec-reference/
+
+# Here the instructions define your application as a service called "server".
+# This service is built from the Dockerfile in the current directory.
+# You can add other services your application may depend on here, such as a
+# database or a cache. For examples, see the Awesome Compose repository:
+# https://github.com/docker/awesome-compose
+services:
+  backend:
+    build:
+      context: ./backend
+    env_file: ./backend/.env
+    environment:
+      - NODE_ENV=dev
+      - MONGO_URI=mongodb://mongo:27017
+    ports:
+      - 3000:3000
+    command: node --watch-path ./src index.js
+    volumes:
+      - ./backend/src:/usr/src/app/src
+      - ./backend/tests:/usr/src/app/tests
+      - ./backend/index.js:/usr/src/app/index.js
+
+  frontend:
+    build:
+      context: ./frontend
+    env_file: ./frontend/.env
+    environment:
+      - NODE_ENV=dev
+    ports:
+      - 5173:5173
+    volumes:
+      - ./frontend/public:/usr/src/app/public
+      - ./frontend/src:/usr/src/app/src
+      - ./frontend/index.html:/usr/src/app/index.html
+      - ./frontend/vite.config.js:/usr/src/app/vite.config.js
+
+  mongo:
+    image: mongo:latest
+    ports:
+      - "27017:27017"
+    environment:
+      - MONGO_INITDB_DATABASE=cdj
+    volumes:
+      - mongo-data:/data/db
+
+volumes:
+  mongo-data:
```

mongod is quite verbose. The following suggestions moves the logs to a file instead of printing them to the console and improves readability of Morgan logs.

```suggestion
  mongo:
    image: mongo:latest
    command: ["mongod", "--logpath", "/var/log/mongodb/mongod.log", "--logappend"]
    ports:
      - "27017:27017"
    environment:
      - MONGO_INITDB_DATABASE=cdj
    volumes:
      - mongo-data:/data/db
      - mongo-log:/var/log/mongodb
    logging:
      driver: json-file
      options:
        max-size: "200m"
        max-file: "10"

volumes:
  mongo-data:
  mongo-log:
```

---

Sure I think it's a good suggestion. I think it makes sense to not use the --logappend option because this way there's a new log file for each time you start the application, so there's cleaner log rotation

---

I agree. Removing `--logappend` makes sense especially with a 10 file max. It would be much cleaner and easier to identify which session something went wrong in too instead of trying to parse through one long file.

---

Just pushed the commit with the change. Resolving converasation